src/can-observable-array.js

```diff
@@ -60,6 +62,24 @@ class ObservableArray extends MixedInArray {
 		return new this(...array);
 	}
 
+	static of (type) {
+		const name = `ObservableArrayOf${canReflect.getName(type)}`;
+		let ObservableArrayOfClass;
+		if (ObservableArrayOfCache.has(type)) {
+			ObservableArrayOfClass = ObservableArrayOfCache.get(type);
+		} else {
+			ObservableArrayOfClass = class ObservableArrayOfClass extends ObservableArray {
+				static get items () { 
+					return { type };
+				}
+			};
+			// Set the name of the class
+			canReflect.setName(ObservableArrayOfClass, name);
+			ObservableArrayOfCache.set(type, ObservableArrayOfClass);
+		}
+		return new ObservableArrayOfClass();
```

Comment:
This shouldn't call the function. We want to be able to use the constructor as a `type`.

Reply:
Ah.. I see. Gotcha